components/brave_wallet/common/fil_address.cc

```diff
@@ -107,8 +119,18 @@ FilAddress FilAddress::FromAddress(const std::string& address) {
     return FilAddress();
   }
 
-  std::string payload_decoded{
-      base32::Base32Decode(base::ToUpperASCII(address.substr(2)))};
+  std::string payload_decoded;
+  if (protocol == mojom::FilecoinAddressProtocol::DELEGATED) {
+    if (address.substr(2, 3) != "10f") {
+      return FilAddress();
+    }
+    payload_decoded =
+        base32::Base32Decode(base::ToUpperASCII(address.substr(5)));
+  } else {
+    payload_decoded =
+        base32::Base32Decode(base::ToUpperASCII(address.substr(2)));
```

Comment:
Does it actually needs ToUpper? https://source.chromium.org/chromium/chromium/src/+/main:components/base32/base32.cc;l=23-28?q=ReverseMapping&ss=chromium

Reply:
Seems letter case is taken into account